tests/identify_udev_tests.c

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for license
+ * information.
+ */
+
+#include <ctype.h>
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// clang-format off
+#include <cmocka.h>
+// clang-format on
+
+#include "capture.h"
+#include "identify_udev.h"
+#include "nvme.h"
+
+char *__wrap_nvme_identify_namespace_vs_for_namespace_device(const char *namespace_path)
+{
+    check_expected_ptr(namespace_path);
+    return mock_ptr_type(char *);
+}
+
+static int setup(void **state)
+{
+    capture_setup(state);
+    unsetenv("DEVNAME");
+    return 0;
+}
+
+static int teardown(void **state)
+{
+    capture_teardown(state);
+    unsetenv("DEVNAME");
+    return 0;
+}
+
+static void test_print_udev_key_value(void **state)
+{
+    (void)state; // Unused parameter
+
+    print_udev_key_value("type", "local");
+
+    assert_string_equal(capture_stdout(), "AZURE_DISK_TYPE=local\n");
+    assert_string_equal(capture_stderr(), "");
+}
+
+static void test_print_udev_key_values_for_vs_success(void **state)
+{
+    (void)state; // Unused parameter
+
+    char vs[] = "type=local,index=2,name=nvme-600G-2";
+
+    int result = print_udev_key_values_for_vs(vs);
+
+    assert_int_equal(result, 0);
+    assert_string_equal(capture_stdout(), "AZURE_DISK_TYPE=local\n"
+                                          "AZURE_DISK_INDEX=2\n"
+                                          "AZURE_DISK_NAME=nvme-600G-2\n");
+    assert_string_equal(capture_stderr(), "");
+}
+
+static void test_print_udev_key_values_for_vs_failure(void **state)
```

Comment:
One other test case you might consider is weirder input like nothing but commas, multiple `=` characters, stuff like that.